tests/test_utils.py

```diff
@@ -90,6 +90,24 @@ def test_safe_padded_read_padding_formats():
         assert region.shape == (8 + 2, 8 + 2)
 
 
+def test_safe_padded_read_padding_shape():
+    """Test safe_padded_read for padding shape."""
+    data = np.zeros((16, 16))
+
+    bounds = (1, 1, 5, 5)
+    with pytest.raises(ValueError):
+        utils.image.safe_padded_read(data, bounds, padding=(1, 1, 1))
+
+
+def test_safe_padded_read_stride_shape():
+    """Test safe_padded_read for padding size."""
+    data = np.zeros((16, 16))
+
+    bounds = (1, 1, 5, 5)
+    with pytest.raises(ValueError):
+        utils.image.safe_padded_read(data, bounds, stride=(1, 1, 1))
```

Comment:
Can we make this into randomsize read test along boundary of 4 edges ? Also, I don't see you put size check on the output. If this funtion fails, error wont be detected in wsi read region because we have auto resize following it.

Reply:
Oh I didn't write these tests but sure can do that.